common/types/ballot.go

```diff
@@ -68,6 +68,8 @@ type Ballot struct {
 	ballotID BallotID
 	// malicious is set to true if smesher that produced this ballot is known to be malicious.
 	malicious bool
+	// beacon that this ballot references either via EpochData or RefBallot
+	beacon Beacon
```

Comment:
i think it is used only with proposal. i think we should set this value on *Proposals rather, then ballot

Reply:
Moved.